fvm/env.go

```diff
@@ -70,10 +125,320 @@ func (env *commonEnv) seedRNG(header *flow.Header) {
 	env.rng = rand.New(source)
 }
 
+// UnsafeRandom returns a random uint64, where the process of random number derivation is not cryptographically
+// secure.
+func (env *commonEnv) UnsafeRandom() (uint64, error) {
+	if env.isTraceable() && env.ctx.ExtensiveTracing {
+		sp := env.ctx.Tracer.StartSpanFromParent(env.traceSpan, trace.FVMEnvUnsafeRandom)
+		defer sp.Finish()
+	}
+
+	if env.rng == nil {
+		return 0, errors.NewOperationNotSupportedError("UnsafeRandom")
+	}
+
+	// TODO (ramtin) return errors this assumption that this always succeeds might not be true
+	buf := make([]byte, 8)
+	_, _ = env.rng.Read(buf) // Always succeeds, no need to check error
+	return binary.LittleEndian.Uint64(buf), nil
+}
+
 func (env *commonEnv) isTraceable() bool {
 	return env.ctx.Tracer != nil && env.traceSpan != nil
 }
 
+func (env *commonEnv) GenerateUUID() (uint64, error) {
```

Comment:
There were slight differences between the two GenerateUUID implementation.  ScriptEnv's check for nil first before metering whereas TransactionEnv meter first before checking for nil.  I've pick the ScriptEnv's implementation as the common implementation since that seem more correct.